src/benchmark.ts

```diff
@@ -36,80 +45,91 @@ const accountB: CreateAccount = {
+      id: BigInt(count),
+      reserved: Buffer.alloc(32, 0),
+      code: 0,
+      flags: 0,
+      timestamp: 0n,
+    })
+  }
+
+  transfers.push(transferBatch)
+  commits.push(commitBatch)
 }
+assert(count === MAX_TRANSFERS)
 
 const main = async () => {
   console.log("creating the accounts...")
   await client.createAccounts([accountA, accountB])
+  const accountResults = await client.lookupAccounts([accountA.id, accountB.id])
+  assert(accountResults.length === 2)
   
-  console.log(`starting benchmark. MAX_TRANSFERS=${MAX_TRANSFERS} REQUEST_BATCH_SIZE=${REQUEST_BATCH_SIZE}`)
+  console.log(`starting benchmark. MAX_TRANSFERS=${MAX_TRANSFERS} REQUEST_BATCH_SIZE=${MAX_REQUEST_BATCH_SIZE} NUMBER_OF_BATCHES=${transfers.length}`)
   let maxCreateTransfersLatency = 0
   let maxCommitTransfersLatency = 0
-  const start = new Date().getTime()
+  const start = Date.now()
 
-  let offset = 0
-  while (offset < transfers.length) {
-    const ms1 = new Date().getTime()
+  for(let i = 0; i < transfers.length; i++) {
+    const ms1 = Date.now()
 
-    // create the batch
-    const transferBatch = transfers.slice(offset, offset + REQUEST_BATCH_SIZE)
-    await client.createTransfers(transferBatch)
+    const transferResults = await client.createTransfers(transfers[i])
+    assert(transferResults.length === 0)
 
-    const ms2 = new Date().getTime()
+    const ms2 = Date.now()
     const createTransferLatency = ms2 - ms1
     if (createTransferLatency > maxCreateTransfersLatency) {
       maxCreateTransfersLatency = createTransferLatency
     }
 
-    // commit the batch
-    const commitBatch = commits.slice(offset, offset + REQUEST_BATCH_SIZE)
-    await client.commitTransfers(commitBatch)
+    const commitResults = await client.commitTransfers(commits[i])
+    assert(commitResults.length === 0)
 
-    const ms3 = new Date().getTime()
+    const ms3 = Date.now()
     const commitTransferLatency = ms3 - ms2
     if (commitTransferLatency > maxCommitTransfersLatency) {
       maxCommitTransfersLatency = commitTransferLatency
     }
-
-    offset += transferBatch.length
-    if (offset % 100000 === 0) {
-      console.log(`${offset} transfers...`)
-    }
   }
 
-  const ms = new Date().getTime() - start
+  const ms = Date.now() - start
   const accounts = await client.lookupAccounts([accountA.id, accountB.id])
-  console.log(accounts)
   console.log("=============================")
-  console.log(`transfers per second: ${Math.floor((1000 * transfers.length)/ms)}`)
+  console.log(`transfers per second: ${Math.floor((1000 * MAX_TRANSFERS)/ms)}`)
   console.log(`create transfers max p100 latency per 10 000 transfers = ${maxCreateTransfersLatency}ms`)
   console.log(`commit transfers max p100 latency per 10 000 transfers = ${maxCommitTransfersLatency}ms`)
-
-  await client.destroy()
+  assert(accounts.length === 2)
+  assert(accounts[0].debits_accepted === BigInt(MAX_TRANSFERS))
+  assert(accounts[1].credits_accepted === BigInt(MAX_TRANSFERS))
```

Comment:
Brilliant checks to have. Can just imagine benchmarking something that wasn't doing the work we thought it was doing. This must be the safest benchmark ever!